agent/pipeline_uploader.go

```diff
@@ -30,14 +30,31 @@ type PipelineUploader struct {
 	RetrySleepFunc func(time.Duration)
 }
 
+// AsyncUploadFlow will first attempt to perform an async pipeline upload and depending
+// on the API's response it will poll for the upload's status.
+//
+// There are 3 "routes" that are relevant
+// 1. Async Route:  /jobs/:job_uuid/pipelines?async=true
+// 2. Upload Route: /jobs/:job_uuid/pipelines
```

Comment:
"Sync" route? (I'm thinking that "Upload" is also what the Async Route does)

Reply:
It's hard to name things.